integration/test-network/installFPC.sh

```diff
@@ -42,33 +42,36 @@ VERBOSE=${VERBOSE:-}
 
 # Install ERCC and ECC
 
+# Org 1
 setGlobals 1
 peer lifecycle chaincode install $PKG_PATH/ercc.tgz
 peer lifecycle chaincode install $PKG_PATH/ecc.tgz
 peer lifecycle chaincode queryinstalled
+ORG1_ALL_INSTALLED=/tmp/installed_chaincodes.org1
+peer lifecycle chaincode queryinstalled > ${ORG1_ALL_INSTALLED}
+ORG1_ERCC_PKG_ID=$(cat ${ORG1_ALL_INSTALLED} | grep ercc | awk '{print $3}' | sed 's/.$//')
+ORG1_ECC_PKG_ID=$(cat ${ORG1_ALL_INSTALLED}  | grep ecc | awk '{print $3}' | sed 's/.$//')
 
+# Org 2
 setGlobals 2
 peer lifecycle chaincode install $PKG_PATH/ercc.tgz
 peer lifecycle chaincode install $PKG_PATH/ecc.tgz
 peer lifecycle chaincode queryinstalled
+ORG2_ALL_INSTALLED=/tmp/installed_chaincodes.org2
+peer lifecycle chaincode queryinstalled > ${ORG2_ALL_INSTALLED}
+ORG2_ERCC_PKG_ID=$(cat ${ORG2_ALL_INSTALLED} | grep ercc | awk '{print $3}' | sed 's/.$//')
+ORG2_ECC_PKG_ID=$(cat ${ORG2_ALL_INSTALLED}  | grep ecc | awk '{print $3}' | sed 's/.$//')
 
-# Get chaincode PGK IDs
-
-ALL_INSTALLED=/tmp/installed_chaincodes
-peer lifecycle chaincode queryinstalled > ${ALL_INSTALLED}
-
-PKGID_ERCC=$(cat ${ALL_INSTALLED} | grep ercc | awk '{print $3}' | sed 's/.$//')
-PKGID_ECC=$(cat ${ALL_INSTALLED}  | grep ecc | awk '{print $3}' | sed 's/.$//')
 
 # Approve
 
 setGlobals 2
-peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ercc --signature-policy "${ERCC_EP}" --version 1.0 --package-id ${PKGID_ERCC} --sequence 1
-peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ecc --signature-policy "${ECC_EP}" --version 1.0 --package-id ${PKGID_ECC} --sequence 1
+peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ercc --signature-policy "${ERCC_EP}" --version 1.0 --package-id ${ORG2_ERCC_PKG_ID} --sequence 1
+peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ecc --signature-policy "${ECC_EP}" --version 1.0 --package-id ${ORG2_ECC_PKG_ID} --sequence 1
 
 setGlobals 1
-peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ercc --signature-policy "${ERCC_EP}" --version 1.0 --package-id ${PKGID_ERCC} --sequence 1
-peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ecc --signature-policy "${ECC_EP}" --version 1.0 --package-id ${PKGID_ECC} --sequence 1
+peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ercc --signature-policy "${ERCC_EP}" --version 1.0 --package-id ${ORG1_ERCC_PKG_ID} --sequence 1
+peer lifecycle chaincode approveformyorg -o ${ORDERER} --channelID ${CHANNEL_ID} --name ecc --signature-policy "${ECC_EP}" --version 1.0 --package-id ${ORG1_ECC_PKG_ID} --sequence 1
```

Comment:
Have you double checked that this works even if `${ORG1_ERCC_PKG_ID} != ${ORG2_ERCC_PKG_ID}`. As we discussed, it should work as the pkg should be peer local value. However, I had some issues while playing with it. Since the install script above deploys the same ercc tar.gz, in this test here the PKG Ids for both orgs are the same.

Reply:
yep, behaves the same way when i create org-specific packages via
```
diff --git a/integration/test-network/installFPC.sh b/integration/test-network/installFPC.sh
index e11c8db..0a44ae6 100755
--- a/integration/test-network/installFPC.sh
+++ b/integration/test-network/installFPC.sh
@@ -44,8 +44,8 @@ VERBOSE=${VERBOSE:-}

 # Org 1
 setGlobals 1
-peer lifecycle chaincode install $PKG_PATH/ercc.tgz
-peer lifecycle chaincode install $PKG_PATH/ecc.tgz
+peer lifecycle chaincode install $PKG_PATH/ercc.peer0.org1.example.com.tgz
+peer lifecycle chaincode install $PKG_PATH/ecc.peer0.org1.example.com.tgz
 peer lifecycle chaincode queryinstalled
 ORG1_ALL_INSTALLED=/tmp/installed_chaincodes.org1
 peer lifecycle chaincode queryinstalled > ${ORG1_ALL_INSTALLED}
@@ -54,8 +54,8 @@ ORG1_ECC_PKG_ID=$(cat ${ORG1_ALL_INSTALLED}  | grep ecc | awk '{print $3}' | sed

 # Org 2
 setGlobals 2
-peer lifecycle chaincode install $PKG_PATH/ercc.tgz
-peer lifecycle chaincode install $PKG_PATH/ecc.tgz
+peer lifecycle chaincode install $PKG_PATH/ercc.peer0.org2.example.com.tgz
+peer lifecycle chaincode install $PKG_PATH/ecc.peer0.org2.example.com.tgz
 peer lifecycle chaincode queryinstalled
 ORG2_ALL_INSTALLED=/tmp/installed_chaincodes.org2
 peer lifecycle chaincode queryinstalled > ${ORG2_ALL_INSTALLED}
diff --git a/integration/test-network/package.sh b/integration/test-network/package.sh
index 781183d..11ca868 100755
--- a/integration/test-network/package.sh
+++ b/integration/test-network/package.sh
@@ -27,10 +27,11 @@ packageChaincode() {
   cc_type=$4
   port=$5

+  for p in "${PEERS[@]}"; do
+
   tmp_dir=$(mktemp -d -t "${cc_name}-packageXXX")
   mkdir -p "${output_dir}"

-  for p in "${PEERS[@]}"; do
     local out_dir=${tmp_dir}/connections/${p}

     mkdir -p $out_dir
@@ -39,17 +40,17 @@ packageChaincode() {
     echo ${out_dir}
     ls ${out_dir}
     cat ${out_dir}/connection.json
-  done

   tar -czf "${tmp_dir}/code.tar.gz" -C ${tmp_dir}/connections .

-  output="${output_dir}/${cc_name}.tgz"
+  output="${output_dir}/${cc_name}.${p}.tgz"

   createMetafile ${tmp_dir} ${cc_name} ${cc_version} ${cc_type}

   tar -czf "${output}" -C "${tmp_dir}" code.tar.gz metadata.json

   rm -rf "${tmp_dir}"
+  done
 }

 createMetafile() {

```
which gives different packages
```
export \
  ORG1_ECC_PKG_ID=ecc_1.0:8087b2b003162feb30bb82797f05f861c96ae771f666ae2a59c2c163a34d4809\
  ORG1_ERCC_PKG_ID=ercc_1.0:3eb2aff4ecd019e9d0bd79b37dfa91100d1cbbcc7e40170b47fe2406208d646e\
  ORG2_ECC_PKG_ID=ecc_1.0:91985ebf3d0713669d888715f6a252505a9d617b0855c2eec7859ef370d1d588\
  ORG2_ERCC_PKG_ID=ercc_1.0:188e79e6f294e6e9a389bf5401a102ee074ae4acc0ce5e1ddc456b7542e64294
```
but produces same test output
```
$ ./test
2020/10/28 08:22:32 ============ Populating wallet ============
 [fabsdk/core] 2020/10/28 15:22:32 UTC - cryptosuite.GetDefault -> INFO No default cryptosuite found, using default SW implementation
2020/10/28 08:22:32 --> Create FPC chaincode enclave:
2020/10/28 08:22:32 Prep attestation params: eyJwYXJhbXMiOlsic29tZSBwYXJhbXMiXX0=
2020/10/28 08:22:32 Received credentials from enclave: serialized_attested_data:{[fpc.Attested_Data]:{cc_params:{chaincode_id:"ercc" version:"1.0" sequence:1 channel_id:"mychannel"} host_params:{peer_identity:"\n\x07Org1MSP\x12\nsome bytes"} enclave_vk:"enclaveVKString"}} attestation:"{\"attestation_type\":\"simulated\",\"attestation\":\"MA==\"}"
2020/10/28 08:22:32 Perform attestation to evidence transformation
2020/10/28 08:22:32 Call registerEnclave at ERCC: eyJwYXJhbXMiOlsic29tZSBwYXJhbXMiXX0=
2020/10/28 08:22:34 --> QueryListEnclaveCredentials:
2020/10/28 08:22:34 --> Result:
2020/10/28 08:22:34 --> Invoke FPC chaincode:
2020/10/28 08:22:34 prepping chaincode params: &{myFunction [arg1 arg2 arg3] fake key}
2020/10/28 08:22:34 Failed to Submit transaction: Failed to evaluate: Multiple errors occurred: - Transaction processing for endorser [localhost:9051]: Chaincode status Code: (500) UNKNOWN. Description: not implemented yet - Transaction processing for endorser [localhost:7051]: Chaincode status Code: (500) UNKNOWN. Description: not implemented yet

```